[SDK-5171] Push subscription when clicking a notification from Terminated state (#8031)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/swift/push_notifications/integration.md

## Subscribing to push notifications updates

To access the push notification payloads processed by Braze, use the [`Braze.Notifications.subscribeToUpdates(payloadTypes:_:)`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/notifications-swift.class/subscribetoupdates(payloadtypes:_:)/) method.

> You can use the `payloadTypes` parameter to specify whether you'd like to subscribe to notifications involving push open events, push received events, or both.

Author: I consolidated this section that was copy-pasted across Swift and ObjC to just live outside the tabs

```objc
NSInteger filtersValue = BRZNotificationsPayloadTypeFilter.opened.rawValue | BRZNotificationsPayloadTypeFilter.received.rawValue;
BRZNotificationsPayloadTypeFilter *filters = [[BRZNotificationsPayloadTypeFilter alloc] initWithRawValue: filtersValue];
BRZCancellable *cancellable = [notifications subscribeToUpdatesWithPayloadTypes:filters update:^(BRZNotificationsPayload * _Nonnull payload) {
  NSLog(@"Braze processed notification with title '%@' and body '%@'", payload.title, payload.body);
}];
```

{% alert important %}
Keep in mind, push received events will only trigger for foreground notifications and `content-available` background notifications. It will not trigger for notifications received while terminated or for background notifications without the `content-available` field.
{% endalert %}

{% endtab %}

{% endtabs %}
{% alert note %}
When using the automatic push integration, `subscribeToUpdates(_:)` is the only way to be notified of remote notifications processed by Braze. The `UIAppDelegate` and `UNUserNotificationCenterDelegate` system methods are not called when the notification is automatically processed by Braze.
{% endalert %}

{% alert tip %}

> Create your push notification subscription in `application(_:didFinishLaunchingWithOptions:)` to ensure your subscription is trigger after an end-user taps a notification while your app is in a terminated state.

Reviewer: ```suggestion
Create your push notification subscription in `application(_:didFinishLaunchingWithOptions:)` to ensure your subscription is triggered after an end-user taps a notification while your app is in a terminated state.
```

> If you want your push notification subscription to be triggered upon clicking on a notification while in the terminated state, you should create your subscription in `application(_:didFinishLaunchingWithOptions:)`. This will ensure that the subscription is set up prior to processing the push click.

Reviewer: ```suggestion
Create your push notification subscription in `application(_:didFinishLaunchingWithOptions:)` to ensure your subscription is trigger after an end-user taps a notification while your app is in a terminated state.
```
